src/shogun/lib/any.h

```diff
@@ -200,6 +200,14 @@ namespace shogun
 		virtual void exit_std_vector(size_t* size) = 0;
 		virtual void exit_map(size_t* size) = 0;
 
+		template <typename T>
+		void on(std::atomic<T>* val)
+		{
+			T tmp = val->load();
+			on(std::addressof(tmp));
+			val->store(tmp);
```

Comment:
the `tmp` variable is a copy of the value managed by std::atomic, so need to store the `tmp` value in case it has been changed when visiting on `T*`